### PR TITLE
Ian's changes

### PR DESCRIPTION
**Pull request by nessthehero**
@Sean-BREI @sharissis-brei 

Please review my changes at your own pace. Add comments next to the lines in the commits, or post them here in the thread.

If you're fine with my changes, please approve the review.

Thanks.

### CODE REVIEW

**Comment by sharissis-brei** [comment]
Looks good to me!

**Comment by nessthehero** [comment]
Approve the pull request then!!

**Comment by sharissis-brei** [comment]
There's no "Approve" button, only "Close and comment". Didn't seem like the same thing.

**Comment by nessthehero** [comment]
@sharissis-brei https://help.github.com/articles/approving-a-pull-request-with-required-reviews/

**Comment by nessthehero** [comment]
👀

File: app/assemble/partials/300-javascript/310-writing-and-formatting-javascript.hbs [hunk 1]
```diff
@@ -26,9 +26,10 @@ Formatting the language statements and patterns should follow these basics:
 * Conditional statements go on a new line followed by the opening brace.
 * Else/else go on the same line as the brace.
 * Use type strict checks with === as opposed to == whenever possible.
+* Use += and -= instead of ++ and --.
 {{/markdown}}
 {{#code "javascript"}}
-for (var i = 0, len = arr.length; i < len; i++) {
+for (var i = 0, len = arr.length; i < len; i += 1) {
```
Comment (Sean-BREI) [inline app/assemble/partials/300-javascript/310-writing-and-formatting-javascript.hbs]: As a best practice as well as a performance boost, get the length of the array in use prior to a for loop. Acquiring the length of an array every instance of a loop slows performance and creates a delay.

var len = arr.length;
for (var i = 0; i < len; i += 0) {
    // Do something here...
}

Comment (nessthehero) [inline app/assemble/partials/300-javascript/310-writing-and-formatting-javascript.hbs]: Hmm, isn't the first portion of a for loop only run before the loop begins?

Comment (Sean-BREI) [inline app/assemble/partials/300-javascript/310-writing-and-formatting-javascript.hbs]: Here are two posts on Stackoverflow that interprets this scenario...

http://stackoverflow.com/questions/8452317/do-loops-check-the-array-length-every-time-when-comparing-i-against-array-length

http://stackoverflow.com/questions/17989270/for-loop-performance-storing-array-length-in-a-variable

Comment (nessthehero) [inline app/assemble/partials/300-javascript/310-writing-and-formatting-javascript.hbs]: I made the change and also added those links to the section.

Comment (nessthehero) [inline app/assemble/partials/300-javascript/310-writing-and-formatting-javascript.hbs]: Don't approve yet. I need to build.

Comment (nessthehero) [inline app/assemble/partials/300-javascript/310-writing-and-formatting-javascript.hbs]: Ok, should be good now.

**Comment by nessthehero** [comment]
🙏
🙌

**Pull request merged**